src/firstboot/firstboot.c

```diff
@@ -164,6 +164,18 @@ static int show_menu(char **x, unsigned n_columns, unsigned width, unsigned perc
         return 0;
 }
 
+static bool locale_is_valid_func(const char *name) {
+        int r;
+
+        r = locale_is_valid(name);
+        if (r == -ENOMEM) {
+                (void) log_oom();
+                return false;
+        }
+
+        return r > 0;
```

Comment:
With this patch, on oom, the user will see:
```
Out of memory.
Entered data invalid.
```
I'd prefer to propagate the refactoring all the way to `prompt_loop`, and make it print the error:
```
Validation failed: %m
```

Reply:
This is not addressed. i.e. the is_valid() function prototype should be changed to return "int", and then generate its own log msg if == 0 is returned, but if < 0 is returned assume the function already printed its own log msgs, so that duplicate logging is removed.